runtime/oti/VMHelpers.hpp

```diff
@@ -2111,22 +2202,40 @@ class VM_VMHelpers
  *	If low tagging(bit0 or bit1) failed due to either
  *
  *   a carrier thread winning to mount, we don't need to do anything, since it will be compensated by pre/post mount actions
+ *   if it is pending to be mounted case(another concurrent scanning block the mounting),
  *   another GC thread winning to scan(bit0/bit1,bit0 and bit1 is irrelevant and independent), again don't do anything, and let the winning thread do the work, instead
  */
 	static VMINLINE bool
-	tryWinningConcurrentGCScan(J9VMContinuation *continuation, bool isGlobalGC)
+	tryWinningConcurrentGCScan(J9VMContinuation *continuation, bool isGlobalGC, bool beingMounted, J9VMThread *vmThread, j9object_t continuationObject)
 	{
-		uintptr_t complementGCConcurrentState = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
-		uintptr_t returnedState = J9_GC_CONTINUATION_STATE_CONCURRENT_SCAN_NONE;
+		volatile uintptr_t *localAddr = &continuation->state;
 		do {
-			/* preserve the concurrent GC state for the other type of GC */
-			complementGCConcurrentState = continuation->state & getConcurrentGCMask(!isGlobalGC);
-			returnedState = VM_AtomicSupport::lockCompareExchange(&continuation->state, complementGCConcurrentState, complementGCConcurrentState | getConcurrentGCMask(isGlobalGC));
-		/* if the other GC happened to change its concurrentGC state since us taking a snapshot of their state, we'll have to retry */
-		} while (complementGCConcurrentState != (returnedState & complementGCConcurrentState));
+			uintptr_t oldContinuationState = *localAddr;
+			if (!isStartedFromContinuationState(oldContinuationState) || isFinishedFromContinuationState(oldContinuationState)) {
```

Comment:
For easier reading introduce another helper isActiveFromContinuationState(), used here or anywhere else.

In GC we really don't distinguish between finished and not started, but just if it is active or not. We could even use only one flag 'active' in the state, but for debugging purposes, let's keep both flags explicit and helpers that deal with them.